components/host-dashboard/AddFundsModal.js

```diff
@@ -0,0 +1,302 @@
+                <AmountDetailsLine
+                  value={values.amount || 0}
+                  currency={host.currency}
+                  label={<FormattedMessage id="addfunds.totalAmount" defaultMessage="Funding amount" />}
+                />
+                <AmountDetailsLine
+                  value={hostFee}
+                  currency={host.currency}
+                  label={
+                    <FormattedMessage
+                      id="addfunds.hostFees"
+                      defaultMessage="Host fees ({hostFees})"
+                      values={{ hostFees: `${hostFeePercent}%` }}
+                    />
+                  }
+                />
+                <AmountDetailsLine
+                  value={platformFee}
+                  currency={host.currency}
+                  label={
+                    <FormattedMessage
+                      id="addfunds.platformFees"
+                      defaultMessage="Platform fees ({platformFees})"
+                      values={{ platformFees: `${platformFeePercent}%` }}
+                    />
+                  }
+                />
+                <StyledHr my={2} borderColor="black.300" />
+                <AmountDetailsLine
+                  value={values.amount - hostFee - platformFee}
+                  currency={host.currency}
+                  label={<FormattedMessage id="addfunds.netAmount" defaultMessage="Net amount" />}
+                  isLargeAmount
+                />
+                <P fontSize="12px" lineHeight="18px" color="black.700" mt={2}>
+                  <FormattedMessage
+                    id="addfunds.disclaimer"
+                    defaultMessage="By clicking below, you agree to set aside {amount} in your bank account on behalf of the collective"
+                    values={{ amount: formatCurrency(values.amount, collective.currency) }}
+                  />
+                </P>
+                {error && <MessageBoxGraphqlError error={error} mt={3} fontSize="13px" />}
+              </ModalBody>
+              <ModalFooter isFullWidth>
+                <Flex justifyContent="center" flexWrap="wrap">
+                  <StyledButton
+                    buttonStyle="primary"
+                    mx={2}
+                    mb={1}
+                    minWidth={120}
+                    disabled={!dirty || !isValid}
+                    loading={isSubmitting}
+                  >
+                    <FormattedMessage id="addfunds.submit" defaultMessage="Add Funds" />
+                  </StyledButton>
+                  <StyledButton mx={2} mb={1} minWidth={100} onClick={props.onClose}>
+                    <FormattedMessage id="actions.cancel" defaultMessage="Cancel" />
+                  </StyledButton>
+                </Flex>
+              </ModalFooter>
+            </Form>
+          );
+        }}
+      </Formik>
+    </StyledModal>
+  );
+};
+
+AddFunds.propTypes = {
+  host: PropTypes.shape({
```

Comment:
If this is strictly the same as `collective.host`, perhaps we could move this into the `collective` prop so we can simplify the query.

Reply:
We fetch the data separately, the root query looks a bit like that:

```gql
host {
  hostedCollectives {
    ...
  }
}
```

We could indeed add a `host` field to the nodes of `hostedCollectives` (which wouldn't have a big impact on SQL thanks to loaders) but it's still more data to fetch, and since the host is always the same the trade-of doesn't seem strong to me for one side or the other.